server/datastore/mysql/scripts.go

```diff
@@ -1973,3 +1973,118 @@ WHERE
 
 	return &summary, nil
 }
+
+func (ds *Datastore) ListBatchScriptExecutions(ctx context.Context, filter fleet.BatchExecutionStatusFilter) ([]fleet.BatchExecutionSummary, error) {
+	stmtExecutions := `
+SELECT
+	COUNT(*) as num_targeted,
+	COUNT(bahr.error) as num_incompatible,
+	COUNT(CASE WHEN hsr.exit_code = 0 THEN 1 END) as num_ran,
+	COUNT(CASE WHEN hsr.exit_code > 0 THEN 1 END) as num_errored,
+	COUNT(CASE WHEN hsr.canceled = 1 AND hsr.exit_code IS NULL THEN 1 END) as num_canceled,
+	(
+		COUNT(*) -
+		(
+			COUNT(bahr.error) +
+			COUNT(CASE WHEN hsr.exit_code = 0 THEN 1 END) +
+			COUNT(CASE WHEN hsr.exit_code > 0 THEN 1 END) +
+			COUNT(CASE WHEN hsr.canceled = 1 AND hsr.exit_code IS NULL THEN 1 END)
+		)
+	) AS num_pending,
```

Comment:
confirmed that these aggregations won't be recomputed, so this is an efficient way to calculate the # of hosts that haven't run yet

Reply:
Does this mean we don't need the extra columns on the `batch_activities` table?
```
num_targeted,
num_pending,
num_ran,
num_errored,
num_incompatible,
num_canceled,
```

Reply:
We'll need those for when the activities are canceled or completed, so that we can save the final results and not have to recalculate them any time we show the list of canceled or completed activities.  Especially since we can't guarantee that the data we're calculating these results from will always exist (e.g. if we cleared out the host script results or something)

Reply:
Sounds good